bug/bug.go

```diff
@@ -242,9 +242,19 @@ func readBug(repo repository.ClockedRepo, ref string) (*Bug, error) {
 	return &bug, nil
 }
 
-func RemoveLocalBug(repo repository.ClockedRepo, id entity.Id) error {
-	ref := bugsRefPattern + id.String()
-	return repo.RemoveRef(ref)
+// RemoveLocalBug will remove a local bug from its hash
+func RemoveBug(repo repository.ClockedRepo, id entity.Id) error {
+	refs, err := repo.ListRefs(id.String())
```

Comment:
That looks quite dangerous. Could you add a test that make sure that this function remove only this particular bug and nothing else ? Maybe start with git a repo, fill it with `FillRepo`, add/pick one bug to remove and check the count after.

Reply:
I'm surprised that even works to be honest.

Reply:
this cyclic thing is annoying. You can do something like that instead:
```go
func TestRemove(t *testing.T) {
	repo := repository.CreateTestRepo(false)
	defer repository.CleanupTestRepos(repo)

	repoCache, err := NewRepoCache(repo)
	require.NoError(t, err)
	
	// generate a bunch of bugs
	rene, err := repoCache.NewIdentity("René Descartes", "rene@descartes.fr")
	require.NoError(t, err)

	for i := 0; i < 100; i++ {
		_, _, err := repoCache.NewBugRaw(rene, time.Now().Unix(), "title", fmt.Sprintf("message%v", i), nil, nil)
		require.NoError(t, err)
	}

	// and one more for testing
	b1, _, err := repoCache.NewBugRaw(rene, time.Now().Unix(), "title", "message", nil, nil)
	require.NoError(t, err)
	
	// ...
}
```

Reply:
> I'll just return an error if we find more than 1 reference.

if the bug has been pushed to multiple git remote, it's possible to have it multiple times and it should be removed multiple times.

Reply:
Actually you will need to push to some remote and fetch to have the correct refs in the repo:
```go
func TestRemove(t *testing.T) {
	repo := repository.CreateTestRepo(false)
	remoteA := repository.CreateTestRepo(true)
	remoteB := repository.CreateTestRepo(true)
	defer repository.CleanupTestRepos(repo, remoteA, remoteB)

	err := repo.AddRemote("remoteA", "file://"+remoteA.GetPath())
	require.NoError(t, err)

	err = repo.AddRemote("remoteB", "file://"+remoteB.GetPath())
	require.NoError(t, err)

	repoCache, err := NewRepoCache(repo)
	require.NoError(t, err)

	// generate a bunch of bugs
	rene, err := repoCache.NewIdentity("René Descartes", "rene@descartes.fr")
	require.NoError(t, err)

	for i := 0; i < 100; i++ {
		_, _, err := repoCache.NewBugRaw(rene, time.Now().Unix(), "title", fmt.Sprintf("message%v", i), nil, nil)
		require.NoError(t, err)
	}

	// and one more for testing
	// b1, _, err := repoCache.NewBugRaw(rene, time.Now().Unix(), "title", "message", nil, nil)
	// require.NoError(t, err)

	_, err = repoCache.Push("remoteA")
	require.NoError(t, err)

	_, err = repoCache.Push("remoteB")
	require.NoError(t, err)

	_, err = repoCache.Fetch("remoteA")
	require.NoError(t, err)

	_, err = repoCache.Fetch("remoteB")
	require.NoError(t, err)
	
	// ...
}
```

Reply:
you can use the code I gave you as a test in the `cache` package and use a variation without cache as a test in the `bug` package:
```go
func TestRemove(t *testing.T) {
	repo := repository.CreateTestRepo(false)
	remoteA := repository.CreateTestRepo(true)
	remoteB := repository.CreateTestRepo(true)
	defer repository.CleanupTestRepos(repo, remoteA, remoteB)

	err := repo.AddRemote("remoteA", "file://"+remoteA.GetPath())
	require.NoError(t, err)

	err = repo.AddRemote("remoteB", "file://"+remoteB.GetPath())
	require.NoError(t, err)

	// generate a bunch of bugs
	rene := identity.NewIdentity("René Descartes", "rene@descartes.fr")
	err = rene.Commit(repo)
	require.NoError(t, err)

	for i := 0; i < 100; i++ {
		b := NewBug()
		createOp := NewCreateOp(rene, time.Now().Unix(), "title", fmt.Sprintf("message%v", i), nil)
		b.Append(createOp)
		err = b.Commit(repo)
		require.NoError(t, err)
	}

	// and one more for testing
	b := NewBug()
	createOp := NewCreateOp(rene, time.Now().Unix(), "title", "message", nil)
	b.Append(createOp)
	err = b.Commit(repo)
	require.NoError(t, err)

	_, err = Push(repo, "remoteA")
	require.NoError(t, err)

	_, err = Push(repo, "remoteB")
	require.NoError(t, err)

	_, err = Fetch(repo, "remoteA")
	require.NoError(t, err)

	_, err = Fetch(repo, "remoteB")
	require.NoError(t, err)

	// ...
}
``